### DISCUSSION TITLE
Set Zoom Programmatically

### DISCUSSION THREAD

**Discussion by aziziaziz** [capacitor-community/camera-preview#278] [Q&A]
Hi,

I would like to ask if is it possible to set the zoom programmatically?
I understand that we can pinch to zoom but some of our users using our application are handicapped which they are not able to pinch.

We are planning to add a slider at the camera page so that the users are able to zoom using the slider.
Is that possible?

If there is such a function, could anyone guide me how to do it since I cannot find it anywhere in the documentation.

Thank you in advance!

**Comment by Importeer** [comment]
Hi Aziz,

This is not an option in the API. I created this method by coding it myself. You know how to extend the default code inside @capacitor-community/camera-preview/ios/Plugin?

I dont use it anymore but is was something like below:

// Plugin.m
`CAP_PLUGIN_METHOD(setZoom, CAPPluginReturnPromise);`

// Plugin.swift
```
@objc func setZoom(_ call: CAPPluginCall) {
        guard let zoomAmount = call.getFloat("zoom") else {
            call.reject("failed to set focuspoint, x is missing")
            return
        }

        do {
            try self.cameraController.setZoom(zoom: zoomAmount)
            call.resolve()
        } catch {
            call.reject("failed to set zoom")
        }
}
```

// CameraController.swift
```
func focusPoint(zoom: Float) throws {
        guard let device = self.currentCameraPosition == .rear ? rearCamera : frontCamera else { return }
       do {
              try device.lockForConfiguration()
              defer { device.unlockForConfiguration() }

              device.videoZoomFactor = zoom
       } catch {
              debugPrint(error)
       }
}
```
